utils/src/main/java/de/hellfish/ngrok/utils/HttpRequestSerializer.java

```diff
@@ -9,22 +12,29 @@
 
 public class HttpRequestSerializer {
 
-    public static void writeToOutputStream(HttpRequest request, OutputStream outputStream) throws RuntimeException {
-        ObjectMapper objectMapper = new ObjectMapper();
+    public static void writeToOutputStream(MyHttpRequest request, OutputStream outputStream) throws RuntimeException {
+        JsonFactory jsonFactory = new JsonFactory();
+        jsonFactory.configure(JsonGenerator.Feature.AUTO_CLOSE_TARGET, false);
+        jsonFactory.configure(JsonParser.Feature.AUTO_CLOSE_SOURCE, false);
+        ObjectMapper objectMapper = new ObjectMapper(jsonFactory);
         try {
             objectMapper.writeValue(outputStream, request);
         } catch (IOException e) {
             throw new RuntimeException("Can't write http-request to socket: Client is not available", e);
         }
     }
 
-    public static HttpRequest readFromInputStream(InputStream inputStream) {
-        ObjectMapper objectMapper = new ObjectMapper();
+    public static MyHttpRequest readFromInputStream(InputStream inputStream) {
+        JsonFactory jsonFactory = new JsonFactory();
```

Comment:
Don't init JsonFactory in the convertor body method. Configure it once somewhere in constructor method.